### DISCUSSION TITLE
Who uses TaskCoach, and how?

### DISCUSSION THREAD

**Discussion by Bjl2AtCbordDotCom** [taskcoach/taskcoach#10] [Ideas]
A big "Thank You" to the authors of TaskCoach, and everyone else who has contributed, in some way.

I'm glad to see that TaskCoach lives on, since I use it daily.  Maybe I just missed it (I'm not familiar with GitHub), but, I don't see a clear 'mission statement', about the reason for this revitalization effort.  

If it suits people, perhaps users could post about how and why they use TaskCoach here?

**Comment by Bjl2AtCbordDotCom** [comment]
I'm a software engineer, and I'm required to log my time in a different system (Jira), and also to report on 'what I did yesterday, and what I'm going to do today', every morning - TaskCoach helps me remember.

Unfortunately, I do a lot of task switching.  TaskCoach makes it easy to switch which task I'm working on, many times per day, and still track time accurately on each task.

The flexible layout of nested tasks is a big plus.  I use a separate file per week, with a 'grandparent' task that rolls up my hours for the whole week, and 'parent' tasks for each day of the week, for grouping and summing purposes, and subtasks for each actual task I need to track time on.

The ability to export is important to me, because I do some manual manipulations to the exported CSV file, and then just make one single entry in Jira for all of the time I spent on each 'story' (a construct in Jira), regardless of what day it was on.

I'm mostly happy with the existing functionality, except that a recent Windows Update (KB5039211) has broken the ability to export, because TaskCoach crashes when the File Explorer dialog opens.  (https://sourceforge.net/p/taskcoach/bugs/1726/)

**Reply by wolftune** [reply]
For reference, GitHub is the most popular coordination/hosting "forge" site for collaborative software development. SourceForge is older and still around, but it is no longer popular. Task Coach was managed in terms of version-control with Mercurial which is also fine but much less popular than Git. So, it makes sense that people working to revive Task Coach would use Git. Unfortunately, GitHub is itself a proprietary service that was bought by Microsoft, so it is not ideal from a software-freedom perspective. I prefer Codeberg.org and suggested it, but people decided to stick to GitHub. GitHub does mean more people already have accounts and know it. We could ever move still.

The main issue for revival is that Task Coach is written in Python programming language and was written at the time of Python2 which is basically never used anymore. For years now, Python3 is the main version in use. There are enough changes that it takes some work to shift the code to Python3. There are then some other libraries that Task Coach uses which could be changed to updated ones. If that work can get done, Task Coach will work better with newer systems and it will be practical again to make changes and keep improving and updating it.

I am the one project-manager non-coder from the original team (I joined when Task Coach already did much of what it does now, only a handful of updates happened during my time, though my time with the project is now 12 years long!)

I use Task Coach every day for an adjusted GTD-style personal organizing management etc. I note things with due dates, regular maintenance routines in my life, everything from my work as a music teacher to just home maintenance and family goals and more. I have moved more and more of my *notes* to just text files lately, but nothing else works like Task Coach for organizing tasks and reminders and choosing where to focus my attention. I currently have 866 tasks in my file.

**Comment by freedomsha** [comment]
Hello, I am a museum technician and I use Taskcoach every day for over 10 years now. At the time I was looking for the GTD method. I find it flexible, free and rather complete. I therefore use Taskcoach every day for all the tasks to be performed in my work, it allows me to forget anything, to attack the urgent tasks and to finish them in time. To be able to choose it according to my desires. To create maintenance routines.

I started working on the code it is 2 years old by wanting to learn the python. I did not suspect that this project was so fat and rich. Wolftune replied but it was complicated, it is still. I made a lot of error. I thank Starofrainnight for having been there to advance the project. Currently, WXPYTHON is not necessarily compatible with the latest version of Python (V13), which makes Taskcoach plant. So I embarked on the conversion of Gui to Tkinter this summer.

**Reply by wolftune** [reply]
Wow! I did not know that news about looking into a different GUI framework and still making progress! Thank you thank you thank you… it would be so amazing to get development unblocked finally.

I have some wishes for what could get Task Coach to be that much better in design if we do finally get it updated to a more-maintainable modern codebase. The foundations are still so good, and being able to improve it still would is a dream I still have. I still use it myself all the time…
